### ISSUE TITLE
'Shortest valid path': takes shortest path although there was a (longer path) with an additional explicit mapping

### ISSUE THREAD

**Issue by jonezdotcom** [johnfairh/TMLPersistentContainer#5]
From User Guide: 
> This library builds the graph of data and mapping models, then analyzes it to find the shortest valid path. Shortest means the fewest migrations. Valid means ‘prefer explicit mappings to inferred ones, even if the path is longer’.

Consider the following state:
```
info: Model graph nodes: [Model, Model4, Model2, Model3]
info: Model graph edges: [Model-(e)-> Model2, Model-(i)-> Model3, Model-(i)-> Model4, Model2-(i)-> Model3, Model2-(i)-> Model4, Model3-(e)-> Model4]
info: Looking for path from Model to Model4
info: Found path [Model-(e)-> Model2, Model2-(i)-> Model4]
```

This seems like a bug to me. According to the documentation it should take a longer path if there is an additional explicit mapping available.

**Comment by johnfairh** [comment]
Thanks, definitely a bug.  With six months' fresh eyes the error is pretty clear :/

Workaround for now if needed is to specify the valid migrations explicitly using `ModelVersionOrder.pairList`.

**Comment by jonezdotcom** [comment]
Thanks. Got to say this library as effing awesome and a lifesaver. Makes manual migrations a breeze, thank you for your work.

**Comment by johnfairh** [comment]
Good to hear!

Fix in master + release 1.0.1.
